packages/typespec-test/test/loadtesting_modular/generated/typespec-ts/src/testRunOperations/api/operations.ts

```diff
@@ -168,9 +168,10 @@ export async function _testRunDeserialize(
           name: result.body.certificate?.["name"],
         },
     environmentVariables: result.body["environmentVariables"],
-    errorDetails: !result.body["errorDetails"]
-      ? result.body["errorDetails"]
-      : result.body["errorDetails"].map((p) => ({ message: p["message"] })),
+    errorDetails:
+      result.body["errorDetails"] === undefined
+        ? result.body["errorDetails"]
+        : result.body["errorDetails"].map((p) => ({ message: p["message"] })),
```

Comment:
this is to work around typescript compiler bug of optional chain, see from [playground link](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEoQM5gEIHsAmAnsgN4CwAUMtcgG7BRgCucANgCJxhwDCuIMYAHMmULsH4B+AFylKNBcnxc47YBgDWGGXKqL9uAA5gJINjpBMAtgCNoAbnn7FAegBUb5ABUAFila4QsAIbMhMIMBgyJa20AB03n7IrOHI6mEYEPjIYLhhEQCOTBCsxMD4EODAMMQQiD5KKkrqGgkAkjDIXNwIPqBCyFZMrCaGrCj4LRgANMh1vcgYPrjD2T5wtChwzVigCFEpIAm+KLRsxYNMWMh2NxBgAO4QlcgADF0g2QBsAMyzeykMMBNgk3C4nM4aIdZDE7FBHHpIdR3J4TmkIiZQpNNItgAAvFCgZBBIRwGyESAYUHgxFI7EaADK+IgAHEbDDrHCEZCAL4AbQAuhDkDzKKKKJRQJBYIgUABZAjDOBQPBEXQKeiMFgcFR8ATCUTiKSyci0hTKbhqTTaE3CyFGEz8cwc2JQO3OFGJfyBYKhcKRaKc+Je5KpdJXLI5PL+oolMoVKo1Ob1RrcHatZAdLpgHp9EADIYjYBjCZTWbzBpLFasNYbLY7Ewgfaho4hs4pFBDa63OyPZ4gN4fb5-dEIQHAiDU936aGB13cpHIjwh0CRYBYlq4gno4nCMkUzBTs3OelMgks7AurnTkWC4Xi8WUGDhfamRbQdesZkAChsBEIsgKvgSoqv+ACUsjoFgqrEKaChQPcogDnBziaswbCcNweqCCIYiOiAtrHvoFqqFMsgAIR-kQcRodqmG8PwOGGvhkhxHyABEJFWlo7FCkRSKSDc-40Qw6E6lhjEGnhpisRxXFTLxN6QrIVGECJWoYbqkm4UaICyZxKjcRgvFxFYcCGN+36GGByAALwAHzIN+KGLpCs6GBxhy8dMSmuaezJsrIHmcS0Z6stg3m+YuDqmGwQUcTFTqsJF-G8mBYE3uKCgPkAA)